src/engine/macros.rs

```diff
@@ -82,3 +82,44 @@ macro_rules! set_blockdev_user_info {
         }
     };
 }
+
+macro_rules! create_pool_check_num {
```

Comment:
I generally prefer to avoid macros, as they are slightly more confusing than source. So, if it were possible to make all these functions rather than macros that would be great. The motivation for making them macros would be to share them between sim and strat engine within the existing infrastructure. But we've used the macros only when constrained to, i.e., when functions would not be possible. It would be fine to start a new module in ```src/engine``` to hold any shared functions.

Reply:
Looks like we need macros to handle the literal format string, which is not really surprising, so sticking with that.

Reply:
I'll make the edit for clarity on the `.join()` calls.

Reply:
Resolved.